### DISCUSSION TITLE
Transaction detail window design with extra features.

### DISCUSSION THREAD

**Discussion by Maniixer** [dogecoin/dogecoin#3067] [General]
Would like to hear some feedback on this.

- Added the value of DOGE in Dollars (Converter)
- Removed unnecessary decimals.

- Added 2 links that are already being used in the CORE wallet when right clicking a transaction, it might be more user friendly if it also shows in the details window.

- For the "Converter" i think that it would be a nice feature as you won't have to look up online for every transaction for it.
What i don't know if this is to much to ask for as it might need some complex coding for it. 

If the converter would work it could also be used in the WoW section where you see your main balance, as most wallets have this feature i thought it would be nice if the core wallet had it also :p

![image](https://user-images.githubusercontent.com/92598085/180612551-afa9c53c-6803-45f7-9e7d-1d8c5b52de9e.png)

**Comment by alamshafil** [comment]
I'll try my best to be neutral for adding the DOGE value converter. 

**Coding-wise**
It's not extremely hard to implement this feature. As a matter of fact I actually have code similar to this that I could use for Dogecoin Core. So for coding-wise it's not impossible but there are other things to consider for a core wallet.

**Security-wise**
The Dogecoin Core is the main implementation of the Dogecoin protocol and can act as a node and/or wallet. It's important that Dogecoin Core stays secure and decentralized. The problem adding this feature is that would depend on using an external API from an exchange or a site like CoinGecko or CoinMarketCap. Making Dogecoin dependent on a external API can cause issues and can decrease "decentralized" in a sense since data is being sourced from a "centralized" API. If the API goes down or provides invalid information  can lead to end user blaming the core wallet for not working. There are many other issues with security, for example data is being sent through the internet instead of from the P2P network, etc. Other people are more knowledgeable about this.

**Example Compromise**
I personally have no problem with this feature; I see it being useful for many shibes who use Dogecoin Core as a wallet. It would be difficult to find a compromise between the average shibe and a "hardcore" decentralized shibe. I feel it's important to maintain security and decentralization while being useful for shibes. If shibes **really** want this feature then I have a compromise plan. By default we could disable this feature but it can be enabled from the settings. The user has the ability to choose which API to use or use a custom one (this idea can be complicated). They also have the ability to choose their currency. Before the user enables this feature a warning should appear similar to the one when create a paper wallet.

**Reply by Maniixer** [reply]
Now that i think about it, if it is a risk it would not be a good idea security wise adding this feature, its best to avoid risk possibilities.

**Reply by alamshafil** [reply]
Yeah it's a different situation for Dogecoin Core, since it's the main implementation that acts as a node. We don't want a vulnerability to be exploited (unlikely but possible). However, I don't know much about this; It would be better if we got professional opinions on this matter.
